packages/framework/src/Model/Order/Processing/OrderProcessorMiddleware/ApplyNominalPromoCodeMiddleware.php

```diff
@@ -80,20 +83,28 @@ public function createAndAddOrderItemData(
      * @param \Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCode $promoCode
      * @param \Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCodeLimit\PromoCodeLimit $promoCodeLimit
      * @param \Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig $domainConfig
+     * @param \Shopsys\FrameworkBundle\Component\Money\Money $totalApplicableProductsPrice
      * @return \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemData|null
      */
     protected function createDiscountOrderItemData(
         PromoCode $promoCode,
         PromoCodeLimit $promoCodeLimit,
         DomainConfig $domainConfig,
+        Money $totalApplicableProductsPrice
     ): ?OrderItemData {
         $locale = $domainConfig->getLocale();
         $domainId = $domainConfig->getId();
 
         $defaultVat = $this->vatFacade->getDefaultVatForDomain($domainId);
 
+        $discountValue = Money::create($promoCodeLimit->getDiscount());
+
+        if ($discountValue->isLessThan($totalApplicableProductsPrice)) {
+            $discountValue = $totalApplicableProductsPrice;
+        }
```

Comment:
🤔  I think we've missed the logic flaw here... 
Shouldn't it be `$discount->isGreaterThan($totalApplicableProductsPrice)`? 

Now I have an order completely free for the 500€ nominal coupon 😆 

![Snímek obrazovky 2024-06-19 v 9 13 20](https://github.com/shopsys/shopsys/assets/1177414/c4da8b75-ccc7-4bf8-b3c4-41d4022b7654)

Reply:
Indeed we did. Not it should be ok.